packages/astro/src/assets/runtime.ts

```diff
@@ -0,0 +1,54 @@
+import {
+	createComponent,
+	render,
+	spreadAttributes,
+	unescapeHTML,
+} from '../runtime/server/index.js';
+import type { ImageMetadata } from './types.js';
+import { makeNonEnumerable, normalizeProps } from './utils/svg.js';
+
+export interface SvgComponentProps {
+	meta: ImageMetadata;
+	attributes: Record<string, string>;
+	children: string;
+}
+
+/**
+ * Make sure these IDs are kept on the module-level so they're incremented on a per-page basis
+ */
+let ids = 0;
+export function createSvgComponent({ meta, attributes, children }: SvgComponentProps) {
+	const id = `a:${ids++}`;
+	const rendered = new WeakSet<Response>();
+	const Component = createComponent((result, props) => {
+		const { title: titleProp, viewBox, mode, ...normalizedProps } = normalizeProps(attributes, props);
+		const title = titleProp ? unescapeHTML(`<title>${titleProp}</title>`) : '';
+
+		if (mode === 'sprite') {
+			// On the first render, include the symbol definition
+			let symbol: any = '';
+			if (!rendered.has(result.response)) {
+				// We only need the viewBox on the symbol definition, we can drop it everywhere else
+				symbol = unescapeHTML(`<symbol${spreadAttributes({ viewBox, id })}>${children}</symbol>`);
+				rendered.add(result.response);
+			}
+
+			return render`<svg${spreadAttributes(normalizedProps)}>${title}${symbol}<use href="#${id}" /></svg>`;
+		}
+
+		// Default to inline mode
+		return render`<svg${spreadAttributes({viewBox, ...normalizedProps})}>${title}${unescapeHTML(children)}</svg>`
+	});
+
+	makeNonEnumerable(Component);
```

Comment:
Is there a reason to make non-enumerable here?

Reply:
Looks like it's to prevent revealing that this is a component. I think I'm fine with it, though I wonder if we could make this dev only